src/sys/windows/named_pipe.rs

```diff
@@ -307,6 +307,25 @@ impl Inner {
             Ok(transferred as usize)
         }
     }
+
+    /// Calls the `PeekNamedPipe` function to get the remaining size of message in NamedPipe
+    #[inline]
+    unsafe fn remaining_size(&self) -> io::Result<usize> {
+        let mut remaining = 0;
+        let r = PeekNamedPipe(
+            self.handle.raw(),
+            std::ptr::null_mut(),
+            0,
+            std::ptr::null_mut(),
+            std::ptr::null_mut(),
+            &mut remaining,
```

Comment:
Is it guaranteed that `ERROR_MORE_DATA` is never returned for stream oriented named pipes?

Because according to the `PeekNamedPipe` docs (https://learn.microsoft.com/en-us/windows/win32/api/namedpipeapi/nf-namedpipeapi-peeknamedpipe) this will return 0 for streams. Which would mean we're effectively creating an infinite loop where we try to read using a buffer of length 0.

Reply:
I'd be surprised if `ERROR_MORE_DATA` is ever returned by stream mode named pipe.

But that's a fair point. We can just err out if `PeekNamedPipe` returned 0.

Reply:
In case `PeekNamedPipe` returns 0 we should probably just return the short-read buffer, not an error.

Reply:
That will happen only when it's stream mode, and we do not expect it to happen, no?

I think we should at least have a debug_assert in this case